proto/contimage/contimage.proto

```diff
@@ -49,6 +49,8 @@ message ContainerImage {
     }
     History         history   = 5;
   }
-  repeated ContainerImageLayer layers  = 10;
-  google.protobuf.Timestamp    builtAt = 11;
+  repeated ContainerImageLayer layers = 10;
+
+  google.protobuf.Timestamp builtAt     = 11;
+  google.protobuf.Timestamp publishedAt = 13;
```

Comment:
I can’t remember if it’s mandatory to add the `optional` keyword to ensure backward compatibility. Would you know ?

```suggestion
  optional google.protobuf.Timestamp publishedAt = 13;
```

Reply:
I don't think that we need to make it optional to keep it backward compatible. For example, this PR extended it with additional fields without making them optional: https://github.com/DataDog/agent-payload/pull/232/files.

However, as this field will not be immediately populated by the agent, I believe that it's indeed better to make it optional in order to avoid an additional serialisation ([doc](https://protobuf.dev/programming-guides/proto3/#field-labels))

> optional: An optional field is in one of two possible states:
> * the field is set, and contains a value that was explicitly set or parsed from the wire. It will be serialized to the wire.
> * the field is unset, and will return the default value. It will not be serialized to the wire.

>You can check to see if the value was explicitly set.

Reply:
Updated it in https://github.com/DataDog/agent-payload/pull/253/commits/b9137dffb5098dff0fb545ca4a8d04f227aae2f7.

Reply:
I'm personally in favor of making almost all fields optional. It keeps things more flexible; you never truly know when it's going to be OK to not write a field in the future, even if it seems "required" now. We can use comments and code-side validation to achieve desired behavior.